proxy/io.go

```diff
@@ -25,45 +23,35 @@ func WriteChunks(conn *net.TCPConn, c [][]byte) (n int, err error) {
 	return total, nil
 }
 
-func ReadBytes(conn *net.TCPConn) ([]byte, error) {
-	ret := make([]byte, 0)
-	buf := make([]byte, BUF_SIZE)
-
-	for {
-		n, err := conn.Read(buf)
-		if err != nil {
-			switch err.(type) {
-			case *net.OpError:
-				return nil, errors.New("timed out")
-			default:
-				return nil, err
-			}
-		}
-		ret = append(ret, buf[:n]...)
+func ReadBytes(conn *net.TCPConn, dest []byte) ([]byte, error) {
+	n, err := readBytesInternal(conn, dest)
+	return dest[:n], err
+}
 
-		if n < BUF_SIZE {
-			break
+func readBytesInternal(conn *net.TCPConn, dest []byte) (int, error) {
+	totalRead, err := conn.Read(dest)
```

Comment:
Previously, this loop would run once, except when the entire buffer was read. Decided to remove the unnecessary loop.